client/chat/channel.tsx

```diff
@@ -180,20 +200,20 @@ class UserList extends React.Component {
     width: 0,
     height: 0,
   }
-  _contentRef = React.createRef()
-  _listRef = React.createRef()
+  _contentRef = React.createRef<HTMLDivElement>()
+  _listRef = React.createRef<VirtualizedList>()
 
   componentDidMount() {
     this.updateDimensions()
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps: UserListProps) {
     if (prevProps.users !== this.props.users) {
       this._listRef.current?.recomputeRowHeights()
     }
   }
 
-  getRowHeight = ({ index }) => {
+  getRowHeight = ({ index }: { index: number }) => {
```

Comment:
I'm pretty sure react-virtualized has types for these functions (this one and the one below), might be better to use those types, e.g.

```ts
getRowHeight: VirtualizedRowHeightFunction = ({ index }) => // ...
```

(some of them are definitely just done inline though, in which case you can ignore this advice)

Reply:
I'm not sure if I'm doing something wrong, but I can't find the types for those functions anywhere :/